### PR TITLE
Update EVM to support configurable precompiles and allow contract deployment at the given address

### PR DESCRIPTION
**Pull request by ramtinms**
This PR updates the EVM code to
  - support configurable precompile lists
  - allow deployment of a contract at a given address

### CODE REVIEW

File: core/vm/evm.go [hunk 1]
```diff
@@ -419,20 +414,29 @@ func (c *codeAndHash) Hash() common.Hash {
...
+	address common.Address,
+	typ OpCode,
+) ([]byte, common.Address, uint64, error) {
 	// Depth check execution. Fail if we're trying to execute above the
 	// limit.
 	if evm.depth > int(params.CallCreateDepth) {
 		return nil, common.Address{}, gas, ErrDepth
 	}
-	if !evm.Context.CanTransfer(evm.StateDB, caller.Address(), value) {
+	if value.Sign() > 0 && !evm.Context.CanTransfer(evm.StateDB, caller.Address(), value) {
```
Comment (turbolent) [inline core/vm/evm.go:430]: Is this and the check below needed for the objective of this PR, to make precompiles configurable?
If so, why is it needed? Maybe leave a comment explaining why here and below.
If not, maybe move these changes to a separate PR.

**Review by turbolent — COMMENTED** [review]
Nice!

File: core/vm/evm.go (hunk 1 above)
Comment (ramtinms) [reply core/vm/evm.go:430]: Mostly added for optimization given we do a lot of deployment (for COA's) without any balance, but will move it to another PR

**Comment by ramtinms** [comment]
closing this one for now.

**Comment by ramtinms** [comment]
breaking this PR into smaller ones
